src/storage/src/controller.rs

```diff
@@ -1150,7 +1151,7 @@ mod persist_read_handles {
                             }
                             PersistWorkerCmd::Downgrade(since_frontiers) => {
                                 for (id, frontier) in since_frontiers {
-                                    downgrades.insert(id, frontier);
+                                    downgrades.insert(id, (span.clone(), frontier));
```

Comment:
perhaps if there's already an entry in the map, we want to link them all up with at least follows_from? quite possibly that's overthinking it though

https://docs.rs/tracing/latest/tracing/span/struct.Span.html#method.follows_from
`This relationship differs somewhat from the parent-child relationship: a span may have any number of prior spans, rather than a single one`

Reply:
I can't do it here because "new" frontiers supersede previous ones, so their work is just dropped. (Side note: I'm not sure how correct that is, now that I think about it again. We should maybe take the latest/highest frontier))

I can change it for `Append`, where I "link in" additional spans via `follows_from`, though that's also not ideal. I'm adding a comment that explains it, though.